modules/smtp/smtp.go

```diff
@@ -18,15 +20,28 @@ type Connection struct {
 	Conn net.Conn
 }
 
+// Verify that an SMTP code was returned, and that it is a successful one!
+func VerifySMTPContents(n int, ret []byte) (string, error){
+	s := string(ret[:n])
+	code, err := getSMTPCode(s)
+	if err != nil {
+		return s, err
+	}
+	if code < 200 || code >= 300 {
+		return s, zgrab2.NewScanError(zgrab2.SCAN_APPLICATION_ERROR, fmt.Errorf("SMTP returned error code %d", code))
```

Comment:
Similar comment; if we wanted to vet that it is a valid code, anything returned as a code from the SMTP RFC would qualify as success.  However, I wonder if we even need to be that strict (in case there are non-standard extensions out there)?